### PR TITLE
Refactor/time

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/membrane/time.ex [hunk 1]
```diff
@@ -11,67 +11,26 @@ defmodule Membrane.Time do
...
-  }
+
+  @units [
+    %{plural: :days, singular: :day, abbrev: "d", duration: 86_400_000_000_000},
+    %{plural: :hours, singular: :hour, abbrev: "h", duration: 3_600_000_000_000},
+    %{plural: :minutes, singular: :minute, abbrev: "min", duration: 60_000_000_000},
+    %{plural: :seconds, singular: :second, abbrev: "s", duration: 1_000_000_000},
+    %{plural: :milliseconds, singular: :millisecond, abbrev: "ms", duration: 1_000_000},
+    %{plural: :microseconds, singular: :microsecond, abbrev: "us", duration: 1_000},
+    %{plural: :nanoseconds, singular: :nanosecond, abbrev: "ns", duration: 1}
```
Comment (kanes115) [inline lib/membrane/time.ex]: Would it be possible to express those numbers as multiplications? Like `60 * 60 * 1000 * 1_000 * 1_000` for hours? (to go through each unit)

File: lib/membrane/time.ex [hunk 2]
```diff
@@ -11,67 +11,26 @@ defmodule Membrane.Time do
...
+  ]
 
   @doc """
   Checks whether value is Membrane.Time.t
   """
   defguard is_t(value) when is_integer(value)
```
Comment (kanes115) [inline lib/membrane/time.ex]: Since it's a public guard, I would go for a full `is_time`. It's short anyway.

File: lib/membrane/time.ex [hunk 3]
```diff
@@ -206,281 +153,105 @@ defmodule Membrane.Time do
...
+    def unquote(as_fun_name)(time) when is_t(time) do
+      Ratio./(time, unquote(unit.duration))
+    end
+  end)
 
   defp best_unit(time) do
-    {unit, divisor} = units() |> Enum.find(fn {_unit, divisor} -> time |> rem(divisor) == 0 end)
-
-    {time |> div(divisor), unit}
+    unit = @units |> Enum.find(&(rem(time, &1.duration) == 0))
```
Comment (kanes115) [inline lib/membrane/time.ex]: I know it shouldn't happen but what if it's not found? Maybe we should check that in guard `is_t`? That the integer is at least convertible to nanoseconds?

Comment (mat-hek) [reply lib/membrane/time.ex]: It is always found since the time is represented in nanoseconds (duration of a nanosecond is 1)

Comment (kanes115) [reply lib/membrane/time.ex]: I know, that's what I meant by `I know it shouldn't happen but...`. I just wonder why we don't check that in `is_time` guard

File: lib/membrane/time.ex (hunk 1 above)
Comment (kanes115) [reply lib/membrane/time.ex]: bump @mat-hek

File: lib/membrane/time.ex (hunk 3 above)
Comment (mat-hek) [reply lib/membrane/time.ex]: We do. Each integer satisfies that.

File: lib/membrane/time.ex (hunk 1 above)
Comment (mat-hek) [reply lib/membrane/time.ex]: why?

**Pull request merged**